Homeworks/module14/test_module14.py

```diff
@@ -0,0 +1,77 @@
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+username = "notabot"
+password = "1234"
+
+
+def test_login(browser):
+    browser.get("https://www.demoblaze.com/")
+    login_button = browser.find_element(By.ID, "login2")
+    login_button.click()
+
+    wait = WebDriverWait(browser, timeout=5)
+
+    wait.until(EC.visibility_of_element_located((By.ID, "loginusername")))
+
+    username_field = browser.find_element(By.ID, "loginusername")
+    password_field = browser.find_element(By.ID, "loginpassword")
+    assert username_field.is_displayed()
+    assert password_field.is_displayed()
+
+    username_field.send_keys(username)
+    password_field.send_keys(password)
+
+    login_button = browser.find_element(By.XPATH, "//button[@onclick='logIn()']")
+    login_button.click()
+
+    wait.until(EC.visibility_of_element_located((By.ID, "logout2")))
+    logout_button = browser.find_element(By.ID, "logout2")
+
+    assert logout_button.is_displayed()
+    assert "Welcome " + username in browser.find_element(By.XPATH, "/html/body").text
+
+
+def test_cart(login):
+    price_elems = login.find_elements(By.XPATH, "//h5")
```

Comment:
maybe it would be better use browser fixture and login fixture separately (it looks counterintuitive to operate with login object like this)